Files/warranty_meraki.py

```diff
@@ -0,0 +1,343 @@
+    def pre_process_meraki_data(self, retry=True, timeout=15):
+        # gets a combined list of all the organization ids associated with the meraki access token
+        all_organization_ids = self.get_all_organizations(retry, timeout)
+
+        if all_organization_ids is None:
+            return None
+
+        # gets license for all organizations and adds it to a dictionary
+        # {organization id: days_remaining}
+        all_licence_states = self.get_license_state_all_organizations(all_organization_ids, retry, timeout)
+
+        if all_licence_states is None:
+            return None
+
+        # gets all the devices in each organization and creates a dictionary to find what organization each device is a
+        # part of --> all_devices = all_organization_devices{serial_number: organization_id, serial_number: ...}
+        # all organizations --> {organization_id: serial_number, serial_number, sserial_number}
+
+        all_organization_devices, all_devices = self.get_all_device_serial_numbers(all_organization_ids, retry, timeout)
+
+        if all_organization_devices is None:
+            return None
+
+        if self.debug is True:
+            print
+            print 'Organization IDs: ' + str(all_organization_ids)
+            print 'License Days Remaining: ' + str(all_licence_states)
+            print 'Organization Devices: ' + str(all_organization_devices)
+            print 'All Devices: ' + str(all_devices)
+            if len(all_devices) == 0:
+                print
+                print "[!] There are no Cisco Meraki devices associated with the given access key"
+                print "[!] No purchase order will be created or updated"
+            print
+
+        return all_licence_states, all_organization_devices, all_devices
+
+    def get_license_state_all_organizations(self, all_organization_ids, retry, timeout):
+        all_license_states = {}
+
+        headers = {
+            'X-Cisco-Meraki-API-Key': self.api_key
+        }
+
+        if self.debug is True:
+            print '\t\t[+] Getting license states for all organizations '
+
+        for organization_id in all_organization_ids:
+            try:
+                resp = requests.get(self.url + self.organization_endpoint + '/' + organization_id +
+                                    self.license_state_endpoint, headers=headers, timeout=timeout)
+                result = self.process_get_response(resp, retry)
+            except requests.RequestException as e:
+                self.error_msg(e)
+                return None
+
+            if result is None:
+                return None
+
+            # Checks to see if expiration date field is valid and converts it to days remaining, adds it to a dictionary
+            # with key organization_id and value days remaining on licence
+            if 'expirationDate' in result:
+                if result['expirationDate'] == "N/A":
+                    if result['status'] == 'License Required':
```

Comment:
Both if and else conditions yield the same days_remaining.  Was that the intent?

Reply:
removed unnecessary if statement. its intention was to include an alternate expiration date for devices based on status. ended up not using it